library/L2_HAL/sensors/environment/temperature/test/si7060_test.cpp

```diff
@@ -1,13 +1,211 @@
 #include "L2_HAL/sensors/environment/temperature/si7060.hpp"
 #include "L4_Testing/testing_frameworks.hpp"
 
-namespace sjsu::lpc40xx
+namespace sjsu
 {
 // Uncomment this when can class has been created
 EMIT_ALL_METHODS(Si7060);
 
 TEST_CASE("Si7060", "[si7060]")
 {
-  SECTION("Initialize") {}
+  constexpr uint8_t kDeviceAddress = Si7060::kDefaultAddress;
+  Mock<I2c> mock_i2c;
+  Si7060 temperature_sensor(mock_i2c.get(), kDeviceAddress);
+
+  SECTION("Initialize")
+  {
+    SECTION("I2c initialization failure")
+    {
+      mock_i2c.Reset();
+      // The failure status should be returned immediately without performing
+      // any further communication with the device.
+      constexpr Status kExpectedStatus = Status::kBusError;
+      bool i2c_transaction_not_invoked = true;
+
+      When(Method(mock_i2c, Initialize)).AlwaysReturn(kExpectedStatus);
+      When(Method(mock_i2c, Transaction))
+          .AlwaysDo([&](I2c::Transaction_t) -> Status {
+            i2c_transaction_not_invoked = false;
+            return Status::kSuccess;
+          });
+
+      CHECK(temperature_sensor.Initialize() == kExpectedStatus);
+      CHECK(i2c_transaction_not_invoked);
+      Verify(Method(mock_i2c, Initialize)).Once();
+      Verify(Method(mock_i2c, Transaction)).Never();
+    }
+    SECTION("incorrect device information")
+    {
+      mock_i2c.Reset();
+      constexpr Status kExpectedStatus     = Status::kDeviceNotFound;
+      constexpr uint8_t kIncorrectDeviceId = 0xFF;
+
+      When(Method(mock_i2c, Initialize)).AlwaysReturn(Status::kSuccess);
+      When(Method(mock_i2c, Transaction))
+          .AlwaysDo([=](I2c::Transaction_t transaction) -> Status {
```

Comment:
Why are you capturing here? You are not using any of the variables here. If you are not using any captured values remove the "=". Rather than using the grab all scoped variables by value using "=" simply list all of the variables you need to capture.
```suggestion
          .AlwaysDo([](I2c::Transaction_t transaction) -> Status {
```